### ISSUE TITLE
fman displays "files" not shown in Explorer

### ISSUE THREAD

**Issue by mherrmann** [fman-community/fman#165]
Explorer does not display certain files, even when "show hidden files" is activated. For instance:

- `ntuser.dat.LOG1`
- `ntuser.dat.LOG2`
- `NTUSER.DAT{...}.TxR.0.regtrans-ms`
- `NTUSER.DAT{...}.TxR.blf`
- `NTUSER.DAT{...}.TM.blf`
- `NTUSER.DAT{...}.TMContainer00000...1.regtrans-ms`

These files seem to be created by the Windows Registry, eg. for backup purposes.

The question is: Should fman display these files or not? Technically, they are files.

**Comment by mherrmann** [comment]
Windows:
![windows](https://cloud.githubusercontent.com/assets/1076393/26547731/c00ce064-4470-11e7-8fe7-4cc79f8afa32.png)

fman:
![fman](https://cloud.githubusercontent.com/assets/1076393/26547732/c2590f00-4470-11e7-8517-8453118ff252.png)

**Comment by kek91** [comment]
I don't think this should be considered a bug, I'm glad fman shows them, but you might want to separate them as hidden files.

These files has a special "System" attribute and that's why they're invisible in Windows Explorer (unless you manually uncheck `Hide protected operating system files (Recommended)` in Explorer settings).

I don't think it makes any sense to create an additional "Show system files" in fman (unless there's some equvialent in Linux and macOS as well), but I think these files could be classified as hidden so only show them if "Show hidden files" are enabled in fman. Just my two cents :)

Edit: when I think about it, isn't this already the default in fman? Seems so to me at least, and in my opinion I'd vote to keep it as it is. Interesting to hear other thoughts as always though

**Comment by mherrmann** [comment]
Very interesting observation. fman currently treats them (ie. System files) like hidden files. I also think that's sensible. But one user didn't like it.

**Comment by cr101** [comment]
@mherrmann Those files are unnecessarily polluting the viewer. Personally, I would never use/look for those files and it makes harder to find the files I'm actually looking for.

**Comment by joshas** [comment]
On Windows 7 fman works as expected - those files in user directory are hidden, after turning on "Show hidden files" they become visible. As files have hidden attribute set - no issue here.
fman should not work exactly as Explorer, because it is more oriented to power users who usually need to mess with system files.

**Comment by mherrmann** [comment]
@cr101 but then suppose you or someone else, some day does want to look at the files. If fman hides them even when "Show hidden files" is enabled, you wouldn't be able to access them / let alone know they're there at all?

**Comment by cr101** [comment]
@mherrmann I understand that some folks don't want fman to behave like Windows Explorer but at the same time, fman needs to find a balance between being different and not cluttering the view at the same time.

**Comment by mherrmann** [comment]
I think I can think of a solution that lets everyone get what they want: Make the behaviour configurable. Currently, fman treats system files like hidden files. With the improvement I envision, that would be the default. But you could also change it so system files are _always_ hidden.

What do you guys think?

**Comment by mherrmann** [comment]
Another solution that just came to me would be to let you define _filters_. Total Commander allows this. Then you can supply file type patterns (eg. `ntuser.dat.*`) that are never displayed.

**Comment by oskretc** [comment]
Here is another use case.

My organization has deployed Palo Alto Traps anti-malware protection software. And this software creates some bait files (look at the picture). If you try to open or delete this files Traps will catch your software as Malware. If you change to hidden files, it works but a few minutes later they are back. 

So fman filters are the only solution I can see to make this files go away.

They are not shown in Win Explorer somehow even though I have set it to show all hidden and system files.

I know this is not on top of the list but you must be able to take some executive decisions :laughing:

![image](https://user-images.githubusercontent.com/4963991/41341471-10b80bce-6efa-11e8-87d8-650c25ef8014.png)

**Comment by mherrmann** [comment]
What if you create a file with a similar name (`!!!!!1234567890.pptx`) and set to hidden and as system file @oskretc? Is that also not shown in Win Explorer?

**Comment by oskretc** [comment]
Yes, the file is displayed in Win Explorer

995.jpg is mine
996.jpg is bait


![image](https://user-images.githubusercontent.com/4963991/41346467-da088d9e-6f06-11e8-8052-82ee1c39c8bb.png)

**Comment by oskretc** [comment]
https://www.paloaltonetworks.com/documentation/41/endpoint/newfeaturesguide/security-features/anti-ransomware-protection

**Comment by mherrmann** [comment]
Interesting. I wonder how this software manages to hide the files in Explorer.
